packages/fluentui/react-northstar/src/components/Input/Input.tsx

```diff
@@ -78,161 +92,214 @@ export interface InputProps extends UIComponentProps, ChildrenComponentProps, Su
   wrapper?: ShorthandValue<BoxProps>;
 }
 
-export interface InputState {
-  value?: InputProps['value'];
-  hasValue?: boolean;
-}
-
 export const inputClassName = 'ui-input';
 export const inputSlotClassNames: InputSlotClassNames = {
   input: `${inputClassName}__input`,
   icon: `${inputClassName}__icon`,
 };
 
-class Input extends AutoControlledComponent<WithAsProp<InputProps>, InputState> {
-  inputRef = React.createRef<HTMLElement>();
+export type InputStylesProps = Required<
+  Pick<InputProps, 'fluid' | 'inverted' | 'inline' | 'disabled' | 'clearable' | 'iconPosition'> & {
+    hasIcon: boolean;
+    hasValue: boolean;
+  }
+>;
+
+const Input: React.FC<WithAsProp<InputProps>> & FluentComponentStaticProps<InputProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Input.displayName, context.telemetry);
+  setStart();
+  const {
+    className,
+    input,
+    type,
+    wrapper,
+    disabled,
+    fluid,
+    inverted,
+    inline,
+    clearable,
+    icon,
+    iconPosition,
+    design,
+    styles,
+    variables,
+  } = props;
+  const inputRef = React.createRef<HTMLElement>();
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(Input.handledProps, props);
+  const [htmlInputProps, restProps] = partitionHTMLProps(unhandledProps);
+  const [value, setValue] = useAutoControlled({
+    defaultValue: props.defaultValue,
+    value: props.value as string,
+    initialValue: '',
+  });
+  const [hasValue, setHasValue] = React.useState<boolean>(!!value && (value as string)?.length !== 0);
```

Comment:
There is no need to have it under the state now, it can be just `const` as it used only for `useStyles()`

Reply:
```suggestion
  const hasValue: boolean = !!value && (value as string)?.length !== 0;
```